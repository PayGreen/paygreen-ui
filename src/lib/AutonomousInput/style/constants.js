import { math } from 'polished';

const fieldHeight = '34px';
const fieldRadius = math(fieldHeight + '/2');

export {
    fieldHeight,
    fieldRadius
};