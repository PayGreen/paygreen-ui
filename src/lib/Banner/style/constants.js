const color = {
    first: {
        none: props => props.theme.wab.white00,
        brand: props => props.theme.color.primary.gradientBase,
        theme: props => props.theme.color[props.colorTheme]['gradientBase']
    },
    second: {
        none: props => props.theme.wab.white00,
        brand: props => props.theme.color.secondary.gradientBase,
        theme: props => props.theme.color[props.colorTheme]['gradientShade']
    }
};

export {
    color
};