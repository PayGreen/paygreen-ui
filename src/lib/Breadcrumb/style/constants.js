import { transparentize } from 'polished';

const textColor = {
    default: {
        original: props => props.theme.wab.grey30,
        reverse: props => transparentize(0.4, props.theme.wab.white00),
    },
    link: {
        original: props => transparentize(0.3, props.theme.color[props.colorTheme].main),
        reverse: props => transparentize(0.4, props.theme.wab.white00),
    },
    hover: {
        original: props => props.theme.color[props.colorTheme].main,
        reverse: props => transparentize(0.05, props.theme.wab.white00),
    }
};

export {
    textColor
};