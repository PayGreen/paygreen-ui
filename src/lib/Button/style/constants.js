const colorCollection = {
    main: props => props.theme.color[props.colorTheme].main,
    white: props => props.theme.wab.white00,
    lightGrey: props => props.theme.wab.white20,
    grey: props => props.theme.wab.grey30,
};

const buttonColors = {
    text: {
        fill: {
            original: {
                enabled: colorCollection.white,
                disabled: colorCollection.lightGrey,
            },
            reverse: {
                enabled: colorCollection.main,
                disabled: colorCollection.grey,
            },
        },
        line: {
            original: {
                enabled: colorCollection.main,
                disabled: colorCollection.grey,
            },
            reverse: {
                enabled: colorCollection.white,
                disabled: colorCollection.grey,
            },
        },
    },
    bg: {
        fill: {
            original: {
                enabled: colorCollection.main,
                disabled: colorCollection.grey,
            },
            reverse: {
                enabled: colorCollection.white,
                disabled: colorCollection.grey,
            },
        },
        line: {
            original: {
                enabled: colorCollection.main,
                disabled: colorCollection.grey,
            },
            reverse: {
                enabled: colorCollection.white,
                disabled: colorCollection.grey,
            },
        },
    },
};

const enableType = props => (props.isDisabled ? 'disabled' : 'enabled');

const opacity = {
    original: 0.5,
    reverse: 0.6,
};

export { buttonColors, enableType, opacity };
