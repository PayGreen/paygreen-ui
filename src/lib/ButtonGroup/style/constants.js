import { math } from 'polished';

const bottomCoeff = 0.75;

const buttonSpace = props => props.resetMargin ? props.theme.space.sm : 0;

const calculateSpace = (space, coeff = 1) => {
    return props => math(props.theme.blockSpace[space] + '*' + coeff + '-' + buttonSpace(props));
};

export {
    bottomCoeff,
    buttonSpace,
    calculateSpace
};