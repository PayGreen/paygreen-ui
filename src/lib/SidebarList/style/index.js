import styled from 'styled-components';

const SidebarListBase = styled.ul`
    box-sizing: border-box;
    margin: 0;
    padding: 0;
    li {
        list-style-type: none;
        margin: ${props => props.theme.line} 0;
        padding: 0;
    }
`;

export { SidebarListBase };
