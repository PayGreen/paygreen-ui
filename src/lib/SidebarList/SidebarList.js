import React from 'react';
import PropTypes from 'prop-types';
import {
    colorThemeOptions,
    colorThemeDefault,
    colorPalletOptions,
    colorPalletDefault,
    colorTypeOptions,
    colorTypeDefault,
    greyOptions,
    greyDefault,
    alignOptions,
    alignDefault,
    fontSizeOptions,
    fontSizeDefault,
} from '../../shared/constants';
import { SidebarListBase } from './style';

const SidebarList = props => {
    return (
        <SidebarListBase {...props}>
            {React.Children.map(props.children, (child, index) =>
                child ? (
                    <li key={index} {...props}>
                        {child}
                    </li>
                ) : null,
            )}
        </SidebarListBase>
    );
};
SidebarList.propTypes = {
    textSize: PropTypes.oneOf(Object.values(fontSizeOptions)),

    colorTheme: PropTypes.oneOf(Object.values(colorThemeOptions)),
    colorPallet: PropTypes.oneOf(Object.values(colorPalletOptions)),
    colorWab: PropTypes.oneOf(Object.values(greyOptions)),
    colorType: PropTypes.oneOf(Object.values(colorTypeOptions)),

    isMain: PropTypes.bool,
    hasHoverBase: PropTypes.bool,
    hasHoverAlternative: PropTypes.bool,
    hasHoverEmphasis: PropTypes.bool,
    align: PropTypes.oneOf(Object.values(alignOptions)),
};

SidebarList.defaultProps = {
    textSize: fontSizeDefault,

    colorTheme: colorThemeDefault,
    colorPallet: colorPalletDefault,
    // colorPallet: colorPalletOptions.wab,
    colorWab: greyDefault,
    colorType: colorTypeDefault,

    isMain: false,
    hasHoverBase: true,
    hasHoverAlternative: true,
    hasHoverEmphasis: false,
    align: alignDefault,
};

export default SidebarList;
