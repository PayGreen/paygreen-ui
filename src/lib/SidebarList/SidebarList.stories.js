import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, radios, boolean } from '@storybook/addon-knobs';
import {
    folder,
    colorPalletOptions,
    colorThemeDefault,
    colorThemeOptions,
    fontSizeOptions,
    greyOptions,
} from '../../shared/constants';
import Link from '../Link/Link';
import SidebarItem from '../SidebarItem/SidebarItem';
import SidebarList from './SidebarList';

const colorThemeLabel = 'Color theme';

storiesOf(folder.nav + folder.sub.sidebarMenu + 'SidebarList', module)
    .addDecorator(withKnobs)
    .add('SidebarList', () => (
        <SidebarList isOpen={boolean('is Open', true)}>
            <a href="#">
                <SidebarItem
                    colorTheme={radios(
                        colorThemeLabel,
                        colorThemeOptions,
                        colorThemeDefault,
                    )}
                    isActive={boolean('Is Active', false)}
                >
                    <Link
                        colorPallet={colorPalletOptions.wab}
                        colorWab={greyOptions.grey60}
                        hasUnderline={false}
                        hasHover={false}
                    >
                        Vue d'ensemble
                    </Link>
                </SidebarItem>
            </a>

            <a href="#">
                <SidebarItem
                    colorTheme={radios(
                        colorThemeLabel,
                        colorThemeOptions,
                        colorThemeDefault,
                    )}
                >
                    <Link
                        colorPallet={colorPalletOptions.wab}
                        colorWab={greyOptions.grey60}
                        hasUnderline={false}
                        hasHover={false}
                    >
                        Vue par projet
                    </Link>
                </SidebarItem>
            </a>
        </SidebarList>
    ));
