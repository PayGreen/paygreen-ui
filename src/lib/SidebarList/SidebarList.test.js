import React from 'react';
import TestRenderer from 'react-test-renderer';
import { ThemeDefault } from '../../theme';
import { iconSizeOptions } from '../../shared/constants';
import { LeafIcon } from '../Icon/Icon';
import Link from '../Link/Link';
import Text from '../Text/Text';
import SidebarItem from '../SidebarItem/SidebarItem';
import SidebarList from './SidebarList';

it('renders without crashing', () => {
    const sidebarList = TestRenderer.create(
        <SidebarList theme={ThemeDefault}>
            <a href="#">
                <SidebarItem theme={ThemeDefault}>
                    <LeafIcon
                        iconSize={iconSizeOptions.lg}
                        theme={ThemeDefault}
                    />

                    <Link
                        theme={ThemeDefault}
                        hasUnderline={false}
                        hasHover={false}
                    >
                        Tree
                    </Link>

                    <Text theme={ThemeDefault}>
                        Lorem ipsum dolor sit amet, consectetur adipiscing elit.
                        Maecenas sit amet accumsan dolor. Nullam fringilla quam
                        leo.
                    </Text>
                </SidebarItem>
            </a>
        </SidebarList>,
    );
    expect(sidebarList.toJSON()).toMatchSnapshot();
});
