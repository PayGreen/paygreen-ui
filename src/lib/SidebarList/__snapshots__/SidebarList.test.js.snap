// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<ul
  className="style__SidebarListBase-sc-b5r7k8-0 VMgvG"
>
  <li>
    <a
      href="#"
    >
      <div
        className="style__SidebarItemBase-sc-1i0hz0r-0 hvRPPB"
        type={null}
      >
        <span
          className="style__LinkBase-sc-1gvcrtp-0 jhcqww"
        >
          Tree
        </span>
      </div>
    </a>
  </li>
  <li>
    <a
      href="#"
    >
      <div
        className="style__SidebarItemBase-sc-1i0hz0r-0 hvRPPB"
        type={null}
      >
        <span
          className="style__LinkBase-sc-1gvcrtp-0 jhcqww"
        >
          Other
        </span>
      </div>
    </a>
  </li>
</ul>
`;
